packages/paste-core/components/switch/src/SwitchKnob.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import type {SwitchKnobProps} from './types';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {CheckboxCheckIcon} from '@twilio-paste/icons/esm/CheckboxCheckIcon';
+import {useTheme} from '@twilio-paste/theme';
+
+const SwitchKnob = React.forwardRef<HTMLDivElement, SwitchKnobProps>(
+  ({element, disabled, switchIsOn, isHovering, height, ...props}, ref) => {
+    const theme = useTheme();
+    const SWITCH_PADDING = theme.space ? theme.space.space20 : '4px';
+
+    const moveSwitchKnob = React.useCallback(() => {
+      if (!disabled) {
+        if (!switchIsOn && isHovering) return 'translateX(5%)';
+        if (switchIsOn && !isHovering) return 'translateX(100%)';
+        if (switchIsOn && isHovering) return 'translateX(95%)';
+      } else if (disabled) {
+        if (!switchIsOn) return 'translate(0%)';
+        if (switchIsOn) return 'translateX(100%)';
+      }
+    }, [switchIsOn, isHovering, disabled]);
```

Comment:
suggestion: I would recommend converting this to a [pure function](https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976) as it becomes easier to test and reason about:

```suggestion
  function getSwitchKnobTransformValue(disabled: boolean, isOn: boolean, isHovering: boolean) {
    // the code you have
  }
  ```

And then define it outside the React component (line 7's closure).

 Then you can use it here as
```js
const transformValue = React.useMemo(() => getSwitchKnobTransformValue(disabled, isOn, isHovering), [disabled, isHovering, isOn]);
```

Now in your test you can just import the function and pass it several arguments to test it works as intended in a very elegant manner without needing a whole component mounting and css reading strategy.